hi-ml-cpath/testhisto/testhisto/datasets/test_panda_tiles.py

```diff
@@ -0,0 +1,40 @@
+from pathlib import Path
+import pytest
+import pandas as pd
+
+from health_cpath.datasets.panda_tiles_dataset import PandaTilesDataset
+from health_cpath.utils.naming import TileKey
+from testhisto.mocks.base_data_generator import MockHistoDataType
+from testhisto.mocks.tiles_generator import MockPandaTilesGenerator
+
+
+@pytest.mark.parametrize("version", [0, 1])
+def test_panda_tiles_dataset(version: int, tmp_path: Path) -> None:
+    """Test the PandaTilesDataset class."""
+
+    _ = MockPandaTilesGenerator(
+        dest_data_path=tmp_path,
+        mock_type=MockHistoDataType.FAKE,
+        n_tiles=4,
+        n_slides=10,
+        n_channels=3,
+        tile_size=28,
+        img_size=224,
+        version=version,
+    )
+    base_df = pd.read_csv(tmp_path / PandaTilesDataset.DEFAULT_CSV_FILENAME).set_index(PandaTilesDataset.TILE_ID_COLUMN)
+    dataset = PandaTilesDataset(root=tmp_path)
+
+    coordinates_columns_v0 = {PandaTilesDataset.TILE_X_COLUMN, PandaTilesDataset.TILE_Y_COLUMN}
+    coordinates_columns_v1 = {TileKey.TILE_LEFT, TileKey.TILE_TOP}
+    dataset_columns = set(dataset.dataset_df.columns)
+    base_df_columns = set(base_df.columns)
+
+    assert coordinates_columns_v0.issubset(dataset_columns)   # v0 columns are always present
+
+    if version == 0:
+        assert base_df_columns == dataset_columns
+        assert not coordinates_columns_v1.issubset(dataset_columns)
+    elif version == 1:
+        assert dataset_columns == base_df_columns.union(coordinates_columns_v0)
+        assert not coordinates_columns_v0.issubset(base_df_columns)
```

Comment:
It's tricky to follow what checking here, especially since I'm not sure what version 0/ 1 means. Could you add a comment that is something like the following:
'if version 0, ensure that only the coordinate columns are in the dataset columns. If version 1, check that columns have been added to the original dataframe columns during dataset creation. ' (if I have understood that correctly)

Also - we only check that coordinate_columns_v1 are _not_ in version 0 dataframe. Would we expect that they _are_ in version 1 dataframe and if so, should we assert that too?

Reply:
version 0 and 1 correspond to the old/new tiling. 
Coordinates are present in both versions they are just named differently, v0 we have tile_x and tile_y, v1 we have left and top which is meant to be x and y of v0.

In v1 we just copy them over to x and y because some part of the code base still uses x and y I think

Reply:
+1 on adding more comments. It is extremely helpful to have comments in tests, I'd happily read a whole story.